esmvalcore/preprocessor/_derive/lapserate.py

```diff
@@ -0,0 +1,39 @@
+"""Derivation of variable ``lapserate``."""
+
+import numpy as np
+from iris import Constraint
+
+from ._baseclass import DerivedVariableBase
+
+
+class DerivedVariable(DerivedVariableBase):
+    """Derivation of variable ``lapserate``."""
+
+    @staticmethod
+    def required(project):  # noqa: ARG004
+        """Declare the variables needed for derivation."""
+        return [
+            {"short_name": "ta"},
+            {"short_name": "zg"},
+        ]
+
+    @staticmethod
+    def calculate(cubes):
+        """Calculate the lapse rate.
+
+        Lapse rate = vertical temperature gradient -dT/dz [K km-1].
+
+        """
+        ta_cube = cubes.extract_cube(Constraint(name="air_temperature"))
+        zg_cube = cubes.extract_cube(Constraint(name="geopotential_height"))
+
+        # Lapse rate
+        lapserate_cube = zg_cube.copy()
+        lapserate_cube.data = (
+            np.gradient(ta_cube.core_data(), axis=1)
```

Comment:
it'd be good if we kept this 100% lazy, Dask API tells me there is a `da.gradient` https://docs.dask.org/en/stable/generated/dask.array.gradient.html

Reply:
Cheers, Axel 🍺